dotCMS/src/integration-test/java/com/dotcms/rest/BundleResourceTest.java

```diff
@@ -88,7 +109,114 @@ private void publish(FileInputStream createCOntentFileInputStream) throws DotPub
+            throws DotDataException {
+        final String uuid = UUIDGenerator.generateUuid();
+        final Bundle bundle = new Bundle();
+        bundle.setId(uuid);
+        bundle.setName("testBundle"+System.currentTimeMillis());
+        bundle.setForcePush(false);
+        bundle.setOwner(userId);
+        bundle.setPublishDate(publishDate);
+        APILocator.getBundleAPI().saveBundle(bundle);
+
+        return uuid;
+    }
+
+    /**
+     * BasicAuth
+     */
+    private HttpServletRequest getHttpRequest() {
+        final MockHeaderRequest request = new MockHeaderRequest(
+                new MockSessionRequest(
+                        new MockAttributeRequest(new MockHttpRequest("localhost", "/").request())
+                                .request())
+                        .request());
+
+        request.setHeader("Authorization",
+                "Basic " + new String(Base64.encode("admin@dotcms.com:admin".getBytes())));
+
+        return request;
+    }
+
+    /**
+     * Method to Test: {@link BundleResource#generateBundle(HttpServletRequest, HttpServletResponse, GenerateBundleForm)}
+     * When: Create a bundle and generate the tar.gz file of the given bundle.
+     * Should: Generate the bundle without issues, 200.
+     */
+    @Test
+    public void test_generateBundle_success() throws DotDataException, IOException {
+        //Create new bundle
+        final String bundleId = insertPublishingBundle(adminUser.getUserId(),new Date());
+
+        //Create a Filter since it's needed to generate the bundle
+        createFilter();
+
+        //Create GenerateBundleForm
+        final GenerateBundleForm bundleForm = new GenerateBundleForm.Builder().bundleId(bundleId).build();
+
+        //Call generate endpoint
+        when(response.getOutputStream()).thenReturn(servletOutputStream);
+        final Response responseResource = bundleResource.generateBundle(getHttpRequest(),response,bundleForm);
+
+        Assert.assertEquals(Status.OK.getStatusCode(),responseResource.getStatus());
+        PublisherAPIImpl.class.cast(APILocator.getPublisherAPI()).getFilterDescriptorMap().clear();
+    }
+
+    /**
+     * Method to Test: {@link BundleResource#downloadBundle(HttpServletRequest, HttpServletResponse, String)}
+     * When: Create a bundle and try to download it, but since the tar.gz has not been generated should fail.
+     * Should: return 404 since the file has not been generated
+     */
+    @Test
+    public void test_downloadBundle_fileNotGenerated_return404() throws DotDataException {
+        //Create new bundle
+        final String bundleId = insertPublishingBundle(adminUser.getUserId(),new Date());
+
+        //Call download endpoint
+        final Response responseResource = bundleResource.downloadBundle(getHttpRequest(),response,bundleId);
+
+        Assert.assertEquals(Status.NOT_FOUND.getStatusCode(),responseResource.getStatus());
```

Comment:
Codacy found an issue: [Unnecessary use of fully qualified name 'Assert.assertEquals' due to existing static import 'org.junit.Assert.*'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6239861)